File: 1-10/1. Two Sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    let map = {};
    for (let i = 0; i < nums.length; i++) {
        if (map[target - nums[i]] >= 0) {
            return [map[target - nums[i]], i];
        }
        else {
            map[nums[i]] = i; 
        }
    }
};

File: 1-10/1. Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}
        for i in range(len(nums)):
            remain = target - nums[i]
            if remain in map:
                return [map[remain], i]
            else:
                map[nums[i]] = i
